### ISSUE TITLE
load_model discards the resume epoch: start_epoch is unconditionally reset to 0 after being computed

### ISSUE THREAD

**Issue by mathnathan** [emelex-ai/BRIDGE#226]
## Summary

`TrainingPipeline.load_model` computes the correct resume epoch, logs that it is resuming from
it, and then unconditionally overwrites it with `0` on the next line. Every resumed run
restarts from epoch 0 while reporting that it resumed.

Long-standing, not a regression: present on `5d7e8d9`.

## The code

`bridge/application/training/training_pipeline.py:477-490`:

```python
            # Set the correct starting epoch
            if "epoch" in checkpoint:
                if self.training_config.checkpoint_path and (
                    "pretraining" not in self.training_config.checkpoint_path
                    or "finetuning" not in self.training_config.checkpoint_path
                ):
                    self.start_epoch = checkpoint["epoch"] + 1  # Start from the next epoch
                    self.logger.info(f"Resuming training from epoch {self.start_epoch}")
            else:
                self.logger.warning("Checkpoint doesn't contain epoch information, starting from 0")
                self.start_epoch = 0
            self.start_epoch = 0          # <-- line 488, unconditional, undoes the above

            return True
```

Line 488 has no condition and no comment. It makes the whole `if/else` above it dead as far as
`start_epoch` is concerned; only the log lines survive.

## Impact

`start_epoch` is read once, at `bridge/application/training/training_pipeline.py:427`:

```python
for epoch in range(self.start_epoch, self.training_config.num_epochs):
```

So resuming a 20-epoch run from `model_epoch_9.pth`:

- logs `Resuming training from epoch 10`
- actually loops `range(0, 20)`
- re-runs epochs 0 through 9 against already-trained weights
- overwrites `model_epoch_0.pth` .. `model_epoch_9.pth` via `save_model`
- reports epoch numbers in every metrics row that are 10 lower than the true training age

The weights and optimizer state do load correctly, so training does continue from where it
stopped. What is wrong is the epoch bookkeeping: the loop count, the checkpoint filenames, and
every logged epoch number.

## Suspected history

The guard above it is also suspect and worth deciding on in the same change. The condition

```python
"pretraining" not in path or "finetuning" not in path
```

is true for essentially every path, since a path can rarely contain both substrings. If the
intent was "this is neither a pretraining nor a finetuning checkpoint", it should be `and`, not
`or`. As written the guard admits everything, which may be why the blanket
`self.start_epoch = 0` was added underneath as a workaround rather than fixing the condition.

## Fixing it

1. Delete line 488.
2. Decide what the `pretraining`/`finetuning` guard is meant to express and rewrite it, or drop
   it. If transfer-learning checkpoints genuinely should restart the epoch counter, make that
   explicit with an `else: self.start_epoch = 0` branch rather than an unconditional reset.
3. Add a test: build a checkpoint dict carrying `epoch: 9`, call `load_model`, assert
   `start_epoch == 10`. Also cover the checkpoint-without-`epoch` case, which must stay 0.

Note that `load_model` swallows all exceptions (`except Exception` at line 491) and returns
`False` after setting `start_epoch = 0`, so a test should assert the return value too, or a
silently failed load will look like a successful reset.

## Related

- #223 (`Model._validate_device` and indexless `cuda`)
- #225 (orthographic loss target off-by-one)

**Comment by mathnathan** [comment]
Fixed in `e593b4e`, on `main` via #230.

Left open by mistake: #230's body said `Closes #223, #224, #225, #226, #227`, and GitHub only
parses a closing keyword attached to a single reference, so only #223 closed.

Reproduced first: a checkpoint saved at `epoch=9` records 9 on disk, and a fresh pipeline
pointed at it reported `start_epoch` 0 where 10 was required.

Both parts of the suggested fix landed. The unconditional `self.start_epoch = 0` is gone, and
the guard above it was rewritten rather than dropped. It read

```python
"pretraining" not in path or "finetuning" not in path
```

which is true for every path a checkpoint can realistically have, since one can rarely contain
both words, so it admitted everything and expressed nothing. The three cases are now explicit
in `_set_start_epoch`:

| checkpoint | `start_epoch` |
|---|---|
| no `epoch` key | 0, with a warning |
| path names a pretraining or finetuning artifact | 0, weights seed a **new** run |
| anything else | `checkpoint["epoch"] + 1` |

That middle branch has real referents rather than being invented:
`bridge/infra/data/storage_interface.py` writes to `models/pretraining/`.

Two decisions worth flagging, both slight departures from the issue.

**It keys on `model_path`**, the file actually being loaded, rather than on
`training_config.checkpoint_path`. Previously a direct `load_model(...)` call with no
`checkpoint_path` set could never resume at all. It can now.

**The substring match is over the whole path**, not the filename, because the marker sits in a
directory component. That is fragile in one direction and the docstring says so: an artifacts
directory named `finetuning_runs/` would make every checkpoint under it restart the counter.

Tests cover all three branches plus the case this issue specifically warned about, a corrupt
or missing checkpoint, asserting the **return value** rather than just `start_epoch`, since
`load_model` swallows every exception and a silently failed load otherwise looks exactly like a
successful reset. There is also a test for the consequence rather than the field: a run resumed
at 10 with `num_epochs=12` iterates 2 epochs, not 12.

Note that `run_train_val_loop` has since changed shape in #230, yielding per step rather than
per epoch, but `range(self.start_epoch, num_epochs)` and therefore this behaviour are
unchanged. See `docs/decisions/0006`.
